scripts/endpoints_checker.py

```diff
@@ -2,121 +2,148 @@
-          "af-south-1"];
+endpoints = [
+        "us-east-1", "us-east-2",
+        "us-west-1", "us-west-2",
+        "eu-west-1", "eu-west-2", "eu-west-3", 
+        "eu-central-1", 
+        "eu-north-1",
+        "ap-southeast-1", "ap-southeast-2", 
+        "ap-northeast-1", "ap-northeast-2", "ap-northeast-3", 
+        "ap-south-1", 
+        "ap-east-1",
+        "sa-east-1",
+        "cn-north-1", 
+        "cn-northwest-1",
+        "ca-central-1",
+        "us-gov-west-1","us-gov-east-1",
+        "us-iso-west-1",
+        "me-south-1",
+        "af-south-1"
+        ];
 """
 
-def RemoveCPPComments(text):
+
+def remove_cpp_comments(text):
     def replacer(match):
-        s = match.group(0);
+        s = match.group(0)
         if s.startswith('/'):
-            return " "; # int/**/x=5 -> int x=5, instead of intx=5.
+            return " "  # int/**/x=5 -> int x=5, instead of intx=5.
         else:
-            return s;
-    pattern = re.compile(r'//.*?$|/\*.*?\*/|"(?:\\.|[^\\"])*"', re.DOTALL | re.MULTILINE);
-    return re.sub(pattern, replacer, text);
-
-def SkipFile(fileName):
-    skipFilePattern = re.compile(r'.*source\/model\/BucketLocationConstraint\.cpp'
-            '|.*source\/S3Client\.cpp'
-            '|.*source\/model\/.*Region.*\.cpp'
-            '|.*source\/[^\/]+Endpoint\.cpp'
-            '|.*aws-cpp-sdk-core\/include\/aws\/core/\Region.h'
-            '|.*tests\/.*Test\.cpp'
-            # add more white lists here
-            );
-    if skipFilePattern.match(fileName):
-        return True;
-    return False;
-
-def ScanContent(content):
-    EndpointsPattern = re.compile(r'us-east-1|us-east-2|us-west-1|us-west-2|eu-west-1|eu-west-2|eu-west-3|eu-central-1|eu-north-1|ap-southeast-1|ap-southeast-2|ap-northeast-1|ap-northeast-2|ap-northeast-3|ap-south-1|sa-east-1|sa-east-1|cn-north-1|cn-northwest-1|ca-central-1|us-gov-west-1|us-gov-east-1|me-south-1|af-south-1');
-    return re.search(EndpointsPattern, content);
-
-def CheckFile(inputFile):
-    if SkipFile(inputFile):
-        return False;
-
-    with open(inputFile) as ftarget:
-        content = ftarget.read();
-
-    strippedContent = RemoveCPPComments(content);
-    match = ScanContent(strippedContent);
+            return s
+
+    pattern = re.compile(r'//.*?$|/\*.*?\*/|"(?:\\.|[^\\"])*"', re.DOTALL | re.MULTILINE)
+    return re.sub(pattern, replacer, text)
+
+
+def skip_file(filename):
+    skip_file_pattern = re.compile(r'.*source/model/BucketLocationConstraint.cpp'
+                                   '|.*source/S3Client.cpp'
+                                   '|.*source/model/.*Region.*.cpp'
+                                   '|.*source/[^/]+Endpoint.cpp'
+                                   '|.*aws-cpp-sdk-core/include/aws/core/Region.h'
+                                   '|.*tests/.*Test.cpp'
+                                   # add more white lists here
+                                   )
+    if skip_file_pattern.match(filename):
```

Comment:
could simplify to `return skip_file_pattern.match(filename)`

Reply:
Submitted a new PR with this change and some whitespace adjustments.